File: Sport-Shoes-Api/demo/src/main/java/com/example/demo/model/product/ProductDetail.java
package com.example.demo.model.product;

import com.example.demo.model.order.Oder;
import com.fasterxml.jackson.annotation.JsonBackReference;

import javax.persistence.*;
import java.util.Set;

@Entity
public class ProductDetail {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;
    private int quantity;
    @Column(columnDefinition = "boolean default false")
    private boolean deleteStatus;


    @ManyToOne
    @JoinColumn(name = "product_id", referencedColumnName = "id")
    private Product product;

    @ManyToOne
    @JoinColumn(name = "color_id", referencedColumnName = "id")
    private Color color;

    @ManyToOne
    @JoinColumn(name = "size_id", referencedColumnName = "id")
    private Size size;

    @ManyToOne
    @JoinColumn(name = "category_id", referencedColumnName = "id")
    private Category category;

    @OneToMany(mappedBy = "productDetail")
    private Set<Image> images;

    @OneToMany(mappedBy = "productDetail")
    private Set<Oder> oders;


}

File: Sport_Shoes/fe/src/app/model/product/product-detail.ts
import {Product} from "./product";
import {Oder} from "../order/oder";
import {Color} from "./color";
import {Size} from "./size";
import {Category} from "./category";
import {Image} from "./image";

export interface ProductDetail {
  id?: number;
  quantity?: number;
  deleteStatus?: boolean;
  product?: Product;
  color?: Color;
  size?: Size;
  category?: Category;
  images?: Image[];
  oders: Oder[];
}
